File: templates/rs/sol_part1.rs
use std::env;
use std::fs::File;
use std::io::{self, Read};

fn main() -> io::Result<()> {
    // Check if a file path is provided as a command-line argument
    let args: Vec<String> = env::args().collect();
    if args.len() < 2 {
        eprintln!("Usage: cargo run <file-path>");
        std::process::exit(1);
    }

    // Get the file path from the first command-line argument
    let file_path = &args[1];

    // Open the file
    let mut file = File::open(file_path)?;

    // Read the content of the file into a string
    let mut content = String::new();
    file.read_to_string(&mut content)?;

    // Print the content of the file to the console
    // println!("{}", content);

    Ok(())
}

File: 5/sol_part1.go
package main

import (
	"fmt"
	"io/ioutil"
	"log"
	"os"
)

func main() {
	// Check if a file path is provided as a command-line argument
	if len(os.Args) < 2 {
		log.Fatal("Usage: go run main.go <file-path>")
	}

	// Get the file path from the first command-line argument
	filePath := os.Args[1]

	// Read the content of the file
	content, err := ioutil.ReadFile(filePath)
	if err != nil {
		log.Fatalf("Error reading file: %v", err)
	}

	// Print the content of the file to the console
	// fmt.Println(string(content))
}
